bootstrap/api/v1alpha1/rke2config_types.go

```diff
@@ -17,48 +17,314 @@ limitations under the License.
 package v1alpha1
 
 import (
+	corev1 "k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
 )
 
 // EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
-// Rke2ConfigSpec defines the desired state of Rke2Config
-type Rke2ConfigSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
+// RKE2ConfigSpec defines the desired state of RKE2Config.
+type RKE2ConfigSpec struct {
+	// Files specifies extra files to be passed to user_data upon creation.
+	//+optional
+	Files []File `json:"files,omitempty"`
 
-	// Foo is an example field of Rke2Config. Edit rke2config_types.go to remove/update
-	Foo string `json:"foo,omitempty"`
+	// PreRKE2Commands specifies extra commands to run before rke2 setup runs.
+	//+optional
+	PreRKE2Commands []string `json:"preRKE2Commands,omitempty"`
+
+	// PostRKE2Commands specifies extra commands to run after rke2 setup runs.
+	//+optional
+	PostRKE2Commands []string `json:"postRKE2Commands,omitempty"`
+
+	// AgentConfig specifies configuration for the agent nodes.
+	//+optional
+	AgentConfig RKE2AgentConfig `json:"agentConfig,omitempty"`
+
+	// PrivateRegistriesConfig defines the containerd configuration for private registries and local registry mirrors.
+	//+optional
+	PrivateRegistriesConfig Registry `json:"privateRegistriesConfig,omitempty"`
+
+	// Version specifies the rke2 version.
+	//+optional
+	Version string `json:"version,omitempty"`
+}
+
+type RKE2AgentConfig struct {
+	// RKE2CommonNodeConfig is an inline struct that references common attribtes between agent and server nodes
+	RKE2CommonNodeConfig `json:",inline"`
+}
+
+// RKE2CommonNodeConfig describes some attributes that are common to agent and server nodes
+type RKE2CommonNodeConfig struct {
```

Comment:
Might be good to extract this struct definition to a `types.go` file in this package. So that we keep this file directly for rke2config types.